Organ harvester file cleanup

VladinXXV opened the pull request:
## About The Pull Request
Replaces every single/double letter variable I could find in `code\game\machinery\harvester.dm` with a more descriptive variable name. I hope, anyway.

Also replaces a typecheck section with just writing a better loop.
## Why It's Good For The Game
Uuuuuuh. Code.... readability.... and maintainability? I think?
## Changelog
Not player-facing.

VladinXXV committed:
Replaces some single and double letter variables

Hopefully, this is an improvement.

san7890 commented on code/game/machinery/harvester.dm:
```diff
@@ -75,21 +75,21 @@
-			if(!isitem(A))
+		for(var/occupant_item in carbon_occupant.held_items + carbon_occupant.get_equipped_items())
+			if(!isitem(occupant_item))
 				continue
-			var/obj/item/I = A
-			if(!(HAS_TRAIT(I, TRAIT_NODROP)))
+			var/obj/item/abiotic_item = occupant_item
```
we can do this better, just replace the whole typecheck with something that's written in the forloop

```dm
		for(var/obj/item/abiotic_item in carbon_occupant.held_items + carbon_occupant.get_equipped_items())
```

builtin typecheck is tres bon

san7890 commented on code/game/machinery/harvester.dm:
```diff
@@ -124,21 +124,21 @@
-		if(BP.body_zone != "chest")
-			BP.forceMove(target)    //Move the limbs right next to it, except chest, that's a weird one
-			BP.drop_organs()
+	for(var/obj/item/bodypart/limb_to_remove in operation_order) //first we do non-essential limbs
```
also is `operation_order` guaranteed to just be a list of only types of `/obj/item/bodypart`? if so, add `as anything`

san7890 commented on code/game/machinery/harvester.dm:
```diff
@@ -154,7 +154,7 @@
 		playsound(src, 'sound/machines/microwave/microwave-end.ogg', 100, FALSE)
 
-/obj/machinery/harvester/screwdriver_act(mob/living/user, obj/item/I)
+/obj/machinery/harvester/screwdriver_act(mob/living/user, obj/item/some_screwdriver)
```
what is the var name on the root of `screwdriver_act()`? if it is still `obj/item/I`, please leave it as such so we don't fuck up passing along vars. if it is something else, change it to that.

san7890 commented on code/game/machinery/harvester.dm:
```diff
@@ -164,20 +164,20 @@
 
-/obj/machinery/harvester/crowbar_act(mob/living/user, obj/item/I)
-	if(default_pry_open(I))
+/obj/machinery/harvester/crowbar_act(mob/living/user, obj/item/some_crowbar)
```
ditto check what the root parameter name is for this proc

san7890 reviewed (changes requested):
some code changes that would be greatly appreciated if you could pass in this PR, as well as some nits.

VladinXXV committed:
san7890 Review

Replaces the "some_tool" variables with "tool"
Replaces a typecheck with a built in typecheck
Adds an "as anything" (I checked, operation_order SHOULD only include children of /obj/item/bodypart)
